__tests__/__unit__/uss/ZoweUSSNode.unit.test.ts

```diff
@@ -0,0 +1,617 @@
+        const nodeNoChildren = new ZoweUSSNode(
+            "aDir", vscode.TreeItemCollapsibleState.Collapsed, rootNode, session, "/u", false, profileOne.name, undefined);
+        nodeNoChildren.dirty = false;
+
+        const rootChildren = await nodeNoChildren.getChildren();
+
+        expect(rootChildren.length).toBe(0);
+    });
+
+    it("Tests that error is thrown when node label is blank", async () => {
+        rootNode.label = "";
+        rootNode.dirty = true;
+
+        expect(rootNode.getChildren()).rejects.toEqual(Error("Invalid node"));
+    });
+
+    it("Tests that when bright.List. causes an error on the zowe call, " +
+        "node.getChildren() throws an error and the catch block is reached", async () => {
+            childNode.contextValue = USS_SESSION_CONTEXT;
+            childNode.fullPath = "Throw Error";
+            childNode.dirty = true;
+
+            await childNode.getChildren();
+
+            expect(showErrorMessage.mock.calls.length).toEqual(1);
+            expect(showErrorMessage.mock.calls[0][0]).toEqual("Retrieving response from uss-file-list Error: Throwing an error to check error handling for unit tests!");
+        });
+
+    it("Tests that when bright.List returns an unsuccessful response, " +
+        "node.getChildren() throws an error and the catch block is reached", async () => {
+            childNode.contextValue = USS_SESSION_CONTEXT;
+            childNode.dirty = true;
+            const subNode = new ZoweUSSNode(
+                "Response Fail", vscode.TreeItemCollapsibleState.Collapsed, childNode, null, null, false, profileOne.name, undefined);
+            subNode.fullPath = "THROW ERROR";
+            subNode.dirty = true;
+
+            await subNode.getChildren();
+
+            expect(showErrorMessage.mock.calls.length).toEqual(1);
+            expect(showErrorMessage.mock.calls[0][0]).toEqual("Retrieving response from uss-file-list Error: Throwing an error to check error handling for unit tests!");
+        });
+
+    it("Tests that when passing a session node that is not dirty the node.getChildren() method is exited early", async () => {
+        rootNode.contextValue = USS_SESSION_CONTEXT;
+        rootNode.dirty = false;
+
+        expect(await rootNode.getChildren()).toEqual([]);
+    });
+
+    it("Tests that when passing a session node with no hlq the node.getChildren() method is exited early", async () => {
+        rootNode.contextValue = USS_SESSION_CONTEXT;
+
+        expect(await rootNode.getChildren()).toEqual([]);
+    });
+});
+
+describe("ZoweUSSNode Unit Tests - Function node.openUSS()", () => {
+    let ussNode;
+    let dsNode;
+    const testUSSTree = generateUSSTree([], [ussNode], generateTreeView());
+    defineGlobals("/test/path/");
+
+    testUSSTree.getTreeView.mockReturnValue(generateTreeView());
+    createBasicZosmfSession.mockReturnValue(session);
+
+    Object.defineProperty(Profiles, "getInstance", {
```

Comment:
I don't know if it is possible, Profiles Unit Testing has a global defineProperty for Profiles.getInstance. Do you think you can use that as a pattern?

Reply:
I put it inside the describe() block because someone later might add a new block where getInstance is mocked differently.